homeassistant/components/sunweg/sensor.py

```diff
@@ -0,0 +1,324 @@
+            description=description,
+        )
+        for description in TOTAL_SENSOR_TYPES
+    ]
+
+    # Add sensors for each device in the specified plant.
+    for device in devices:
+        probe = SunWEGData(api, username, password, device.id, "inverter")
+
+        entities.extend(
+            [
+                SunWEGInverter(
+                    probe,
+                    name=f"{device.name}",
+                    unique_id=f"{device.sn}-{description.key}",
+                    description=description,
+                )
+                for description in INVERTER_SENSOR_TYPES
+            ]
+        )
+
+        for phase in device.phases:
+            phase_probe = SunWEGData(
+                api, username, password, device.id, "phase", phase.name
+            )
+            entities.extend(
+                [
+                    SunWEGInverter(
+                        phase_probe,
+                        name=f"{device.name} {phase.name}",
+                        unique_id=f"{device.sn}-{phase.name}-{description.key}",
+                        description=description,
+                    )
+                    for description in PHASE_SENSOR_TYPES
+                ]
+            )
+
+        for mppt in device.mppts:
+            for string in mppt.strings:
+                string_probe = SunWEGData(
+                    api, username, password, device.id, "string", string.name
+                )
+                entities.extend(
+                    [
+                        SunWEGInverter(
+                            string_probe,
+                            name=f"{device.name} {string.name}",
+                            unique_id=f"{device.sn}-{string.name}-{description.key}",
+                            description=description,
+                        )
+                        for description in STRING_SENSOR_TYPES
+                    ]
+                )
+
+    async_add_entities(entities, True)
+
+
+class SunWEGInverter(SensorEntity):
+    """Representation of a SunWEG Sensor."""
+
+    entity_description: SunWEGSensorEntityDescription
+
+    def __init__(
+        self, probe, name, unique_id, description: SunWEGSensorEntityDescription
+    ) -> None:
+        """Initialize a PVOutput sensor."""
+        self.probe = probe
+        self.entity_description = description
+
+        self._attr_name = f"{name} {description.name}"
+        self._attr_unique_id = unique_id
+        self._attr_icon = (
+            description.icon if description.icon is not None else "mdi:solar-power"
+        )
+
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, probe.device_id)},
+            manufacturer="SunWEG",
+            name=name,
+        )
+
+    @property
+    def native_value(
+        self,
+    ) -> (
+        StateType
+        | str
+        | int
+        | float
+        | None
+        | datetime.date
+        | datetime.datetime
+        | Decimal
```

Comment:
StateType = str | int | float | None
Remove unneeded ones

Reply:
Ok. Need datetime as well. I'll remove date and Decimal